compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.cpp

```diff
@@ -938,73 +825,33 @@ MMAScheduleAttr::getContractionLayout(vector::ContractionOp contractOp) const {
     currNCount /= subgroupsUsed;
   }
 
-  SmallVector<int64_t> subgroupBasis;
-  if (batchCount == 1) {
-    subgroupBasis.push_back(1);
-  }
+  SmallVector<int64_t> subgroupMStrides(subgroupMBasis.size());
+  SmallVector<int64_t> subgroupNStrides(subgroupNBasis.size());
+
   auto mDimVec = opInfo.getMDims();
   llvm::SmallDenseSet<int64_t> mDims(mDimVec.begin(), mDimVec.end());
   auto nDimVec = opInfo.getNDims();
   llvm::SmallDenseSet<int64_t> nDims(nDimVec.begin(), nDimVec.end());
-
-  int64_t currM = 0;
-  int64_t currN = 0;
-  // Because we currently require all batch dimensions to be unit, the subgroup
-  // basis can be constructed from the M and N bases. To keep things simple,
-  // the current heuristic is to distribute all M dims followed by all N dims.
-  for (auto dim : llvm::seq(static_cast<int64_t>(0), opInfo.getCRank())) {
-    if (mDims.contains(dim)) {
-      subgroupBasis.push_back(subgroupMBasis[currM]);
-      // Construct mDimVec such that it contains the order in which the M dims
-      // appear in the C matrix.
-      mDimVec[currM] = dim;
-      currM++;
-    }
-    if (nDims.contains(dim)) {
-      subgroupBasis.push_back(subgroupNBasis[currN]);
-      // Construct nDimVec such that it contains the order in which the N dims
-      // appear in the C matrix.
-      nDimVec[currN] = dim;
-      currN++;
-    }
+  // Because we currently require all batch dimensions to be unit, the
+  // subgroup basis can be constructed from the M and N bases. To keep things
+  // simple, the current heuristic is to distribute the loop dimensions from
+  // outer to inner.
+  int64_t currMStride = 1;
+  for (auto [stride, basis] :
+       llvm::reverse(llvm::zip(subgroupMStrides, subgroupMBasis))) {
+    stride = currMStride;
```

Comment:
Does range-loop of some llvm array/vector suppose to loop by reference?

Reply:
References are used by obfuscated by the `zip` using reference proxies.

Shouldn't this also be `zip_equal`, BTW?

Reply:
this code was wrong, removed it